src/ResultWriter/EnergyOutput.cpp

```diff
@@ -0,0 +1,147 @@
+      __attribute__((aligned(PAGESIZE_STACK)));
+  real QInterpolatedMinus[tensor::QInterpolatedMinus::size()]
+      __attribute__((aligned(PAGESIZE_STACK)));
+  real tractionInterpolated[tensor::tractionInterpolated::size()]
+      __attribute__((aligned(ALIGNMENT)));
+
+  krnl.QInterpolated = QInterpolatedPlus;
+  krnl.Q = degreesOfFreedomPlus;
+  krnl.TinvT = godunovData.TinvT;
+  krnl._prefetch.QInterpolated = QInterpolatedPlus;
+  krnl.execute(faceInfo.plusSide, 0);
+
+  krnl.QInterpolated = QInterpolatedMinus;
+  krnl.Q = degreesOfFreedomMinus;
+  krnl.TinvT = godunovData.TinvT;
+  krnl._prefetch.QInterpolated = QInterpolatedMinus;
+  krnl.execute(faceInfo.minusSide, faceInfo.faceRelation);
+
+  dynamicRupture::kernel::computeTractionInterpolated trKrnl;
+  trKrnl.tractionPlusMatrix = godunovData.tractionPlusMatrix;
+  trKrnl.tractionMinusMatrix = godunovData.tractionMinusMatrix;
+  trKrnl.QInterpolatedPlus = QInterpolatedPlus;
+  trKrnl.QInterpolatedMinus = QInterpolatedMinus;
+  trKrnl.tractionInterpolated = tractionInterpolated;
+  trKrnl.execute();
+
+  real staticWork = 0.0;
+  dynamicRupture::kernel::computeFrictionalEnergy feKrnl;
+  feKrnl.slipRateInterpolated = slip;
+  feKrnl.tractionInterpolated = tractionInterpolated;
+  feKrnl.spaceWeights = spaceWeights;
+  feKrnl.frictionalEnergy = &staticWork;
+  feKrnl.timeWeight = -0.5 * godunovData.doubledSurfaceArea;
+  feKrnl.execute();
+
+  return staticWork;
+}
+
+void seissol::writer::printEnergies(GlobalData const* global,
+                                    seissol::initializers::DynamicRupture* dynRup,
+                                    seissol::initializers::LTSTree* dynRupTree,
+                                    MeshReader const& i_meshReader,
+                                    seissol::initializers::LTSTree* i_ltsTree,
+                                    seissol::initializers::LTS* i_lts,
+                                    seissol::initializers::Lut* i_ltsLut,
+                                    bool usePlasticity)
+
+{
+  double totalWorkLocal = 0.0;
+  double staticWorkLocal = 0.0;
+  double plasticMomentLocal = 0.0;
+  for (auto it = dynRupTree->beginLeaf(); it != dynRupTree->endLeaf(); ++it) {
+    /// \todo timeDerivativePlus and timeDerivativeMinus are missing the last timestep.
+    /// (We'd need to send the dofs over the network in order to fix this.)
+    real** timeDerivativePlus = it->var(dynRup->timeDerivativePlus);
+    real** timeDerivativeMinus = it->var(dynRup->timeDerivativeMinus);
+    DRGodunovData* godunovData = it->var(dynRup->godunovData);
+    DRFaceInformation* faceInformation = it->var(dynRup->faceInformation);
+    DROutput* drOutput = it->var(dynRup->drOutput);
+
+#ifdef _OPENMP
+#pragma omp parallel for reduction(+ : totalWorkLocal) reduction(+ : staticWorkLocal)
+#endif
+    for (unsigned i = 0; i < it->getNumberOfCells(); ++i) {
+      if (faceInformation[i].plusSideOnThisRank) {
```

Comment:
Is this to ensure that we do not count duplicate cells twice? Are you sure that this actually works?

Reply:
I don't know. Sounds like a binary partition, right?

Reply:
We should easily be able to verify this with a small mesh and `std::cout` debugging. Or add a counter and count how many elements are processed on every node.

Reply:
done